test/promql/tests/application_errors_test.yaml

```
@@ -0,0 +1,102 @@
+                application-service on cluster cluster01 is failing to
+                successfully delete at least 99% of applications over the past hour
+              runbook_url: TBD
+
+  - interval: 5m
+    input_series:
+
+      # HAS experienced 10% deletion failure rate, should still alert
+      - series: 'has_application_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1+1x99'
+      - series: 'has_application_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '0+1x10 10x88'
+
+    alert_rule_test:
+      - eval_time: 65m
+        alertname: ApplicationDeletionErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: application-service
+              pod: has
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
+                HAS is experiencing application deletion failures of >1%
+              description: >-
+                Application controller in Pod has for namespace
+                application-service on cluster cluster01 is failing to
+                successfully delete at least 99% of applications over the past hour
+              runbook_url: TBD
+
+  - interval: 5m
+    input_series:
+
+      # HAS experienced 5% deletion failure rate, should still alert
+      - series: 'has_application_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1+1x99'
+      - series: 'has_application_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1+1x4'
+
+    alert_rule_test:
+      - eval_time: 65m
+        alertname: ApplicationDeletionErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: application-service
+              pod: has
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
+                HAS is experiencing application deletion failures of >1%
+              description: >-
+                Application controller in Pod has for namespace
+                application-service on cluster cluster01 is failing to
+                successfully delete at least 99% of applications over the past hour
+              runbook_url: TBD
+
+  - interval: 5m
+    input_series:
+
+      # HAS experienced 1% deletion failure rate, should not alert
+      - series: 'has_application_successful_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1+1x98'
+      - series: 'has_application_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1+1x99'
+      - series: 'has_application_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1x99'
```

This test is actually testing zero failure rate if I see it correctly now, as there's no increase in failures during the time period. I think we need something like this:
```suggestion
      - series: 'has_application_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
        values: '0+10x64'
      - series: 'has_application_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
        values: '0x9 5x54'
```
This should give us an increase of 600 total deletion when we look back 1h from 60m or 65m, and it should give us an increase of 5 failures during that same window, which should be just under 1%.